app/models/task.py

```diff
@@ -1,5 +1,34 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import String, DateTime
+from datetime import datetime
+from typing import Optional
+from sqlalchemy import ForeignKey
 from ..db import db
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str] 
+    description: Mapped[str]
+    completed_at: Mapped[datetime | None] = mapped_column(nullable=True)
+
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"), nullable=True)
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
+    
+    def to_dict(self):
+        task_dict = {
+        "id": self.id,
+        "title": self.title,
+        "description": self.description,
+        "is_complete": self.completed_at is not None
```

Comment:
Looks like you are trying to dynamically set this key/value pair to either `True` or `False` based on if `self.completed_at` is truthy or falsy. Though what you have works technically, for readability I would prefer a ternary, or a `bool` cast, to preform this logic as it is the convention. 
```suggestion
        "is_complete": True if self.completed_at else False
```